mysql-test/suite/innodb/t/insert_into_empty.test

```diff
@@ -279,4 +279,21 @@ CREATE TABLE t1(a INT KEY)ENGINE=InnoDB
 INSERT INTO t1 VALUES(1);
 UPDATE t1 SET a = 2 WHERE a = 1;
 DROP TABLE t1;
+
+--echo #
+--echo #  MDEV-34265 Possible hang during IO burst with innodb_flush_sync enabled
+--echo #
+CREATE TABLE t1(f1 MEDIUMTEXT)ENGINE=InnoDB;
+
+SET @save_dbug=@@GLOBAL.debug_dbug;
+SET @@GLOBAL.debug_dbug='+d,ib_page_cleaner_sleep';
+
+SET STATEMENT debug_dbug='+d,ib_free_page_sleep' FOR
+INSERT INTO t1 VALUES(REPEAT(1, 8459264));
+
+SET @@GLOBAL.debug_dbug=@save_dbug;
```

Comment:
These need to be guarded with
```sql
--error 0,1193
```
like elsewhere in the test, or better yet, all occurrences should be changed to
```sql
--error 0,ER_UNKNOWN_SYSTEM_VARIABLE
```
We want this test to be executed also on non-debug builds, to have wide platform coverage.  If needed, you may write like this:
```sql
if ($have_debug) {
SET STATEMENT debug_dbug='+d,ib_free_page_sleep' FOR
INSERT INTO t1 VALUES(REPEAT(1, 8459264));
}
if (!$have_debug) {
--echo SET STATEMENT debug_dbug='+d,ib_free_page_sleep' FOR
INSERT INTO t1 VALUES(REPEAT(1, 8459264));
}
```
The reason why I suggest this trick is that in this way, we can avoid having a `,debug.rdiff` file that could become a maintenance burden.

Reply:
Thanks for pointing it out. I have now made the changes. Please have a look,